modules/sync/admin-sync.php

```diff
@@ -230,4 +245,42 @@ public function __call( $func, $args ) {
 			E_USER_ERROR
 		);
 	}
+
+	/**
+	 * Returns some data (`from_post_id` and `new_lang`) from the current request.
+	 *
+	 * @since 3.7
+	 *
+	 * @param array $post_data A post array.
+	 * @return array
+	 *
+	 * @phpstan-return array{}|array{from_post_id: int<0,max>, new_lang: string}
+	 */
+	public static function get_data_from_request( array $post_data ): array {
+		if ( ! isset( $GLOBALS['pagenow'], $_GET['_wpnonce'], $_GET['from_post'], $_GET['new_lang'], $_GET['post_type'], $post_data['post_type'] ) ) {
+			return array();
+		}
+
+		if ( 'post-new.php' !== $GLOBALS['pagenow'] ) {
+			return array();
+		}
+
+		if ( ! is_string( $_GET['_wpnonce'] ) || ! is_numeric( $_GET['from_post'] ) || ! is_string( $_GET['new_lang'] ) ) {
+			return array();
+		}
+
+		if ( $post_data['post_type'] !== $_GET['post_type'] || ! pll_is_translated_post_type( $post_data['post_type'] ) ) {
```

Comment:
I prefer not to use API functions inside the core classes
```suggestion
		if ( $post_data['post_type'] !== $_GET['post_type'] || ! $this->model->is_translated_post_type( $post_data['post_type'] ) ) {
```

Reply:
Can't use `$this` in a static method, and making the method non-static would make it difficult to reuse.

Reply:
Ok. Let's improve that.
Since we want this method to be called from outside, I would move it outside the class. Indeed, this class is dedicated to hooks and now I prefer not to have classes mixing hooks and "true" public methods.  My proposal is `PLL_Admin_Links` as this is the class where we generate the info read in this method. I would rename it to something like `get_info_from_new_post_translation()` or `get_data_from_new_post_translation_request()` (close to your `pll_...` proposal, or something mixing these...
I would keep it non-static and use `$this->model->is_translated_post_type()` as the other methods in the class. I don't mind if the class is available only from admin side. It's already the case when we generate links.

Reply:
Done in [60d8ed4](https://github.com/polylang/polylang/pull/1532/commits/60d8ed4a7318c9c067bc352837f64dfd8b79ac4e).